src/integration/java/gov/loc/repository/bagit/ExtendedBagVerifier.java

```diff
@@ -0,0 +1,32 @@
+package gov.loc.repository.bagit;
+
+import gov.loc.repository.bagit.domain.Bag;
+import gov.loc.repository.bagit.domain.Manifest;
+import gov.loc.repository.bagit.exceptions.CorruptChecksumException;
+import gov.loc.repository.bagit.exceptions.FileNotInPayloadDirectoryException;
+import gov.loc.repository.bagit.exceptions.InvalidBagitFileFormatException;
+import gov.loc.repository.bagit.exceptions.MaliciousPathException;
+import gov.loc.repository.bagit.exceptions.UnsupportedAlgorithmException;
+import gov.loc.repository.bagit.exceptions.VerificationException;
+import gov.loc.repository.bagit.verify.BagVerifier;
+
+import java.io.IOException;
+
+// an example of extending the BagVerifier in order to add your own checks that suffice the specific
+// verification rules for a bag, as defined by your project
+// NOTE: this cannot override the BagVerifier.isComplete or BagVerifier.isValid, as they're defined
+//       by "The BagIt File Packaging Format" (https://tools.ietf.org/html/draft-kunze-bagit).
+public class ExtendedBagVerifier extends BagVerifier {
```

Comment:
This class doesn't actually test anything. The src/test and src/integration are for junit tests. The integration is meant to test against the bagit-suite as well as the combination of the readers and writers. You should probably put this in the src/test directory following the package name and have junit test something. That way gradle will test it as well as it being tested during compile time.